builtin/update-index.c

```diff
@@ -1098,7 +1099,13 @@ int cmd_update_index(int argc, const char **argv, const char *prefix)
 	argc = parse_options_end(&ctx);
 
 	getline_fn = nul_term_line ? strbuf_getline_nul : strbuf_getline_lf;
+	if (mark_skip_worktree_only && gvfs_config_is_set(-1))
```

Comment:
Do we really want this test to be this strict: do this only when `core_gvfs == -1`? Because in all other cases, `gvfs_config_is_set(-1)` will fail (if I read the source code correctly).

A much gentler way would be `if (core_gvfs && ...)`.

Reply:
core_gvfs isn't referenced anywhere outside of gvfs.c - frankly, I'm not sure why it isn't just static.  Instead, everywhere calls gvfs_config_is_set() so that we can ensure the config setting has been loaded before we access it.  Since gvfs_config_is_set() does a mask with core_gvfs, -1 is the easy way to ensure I'm testing for any bit to be set in core_gvfs.

Reply:
> Since gvfs_config_is_set() does a mask with core_gvfs, -1 is the easy way to ensure I'm testing for any bit to be set in core_gvfs.

Unfortunately, it does not do that. It does not test *any* bit, [instead, it tests for *every* bit](https://github.com/Microsoft/git/blob/5dd0fb7da8aef5dc42d54abd83c98767ddae8fe5/gvfs.c#L34-L38):

```
return (core_gvfs & mask) == mask;
```

where `mask` is `-1`. In other words, if `core_gvfs & -1` (which is the same value as `core_gvfs` as `-1` is "all bits set") is equal to `-1` (i.e. if `core_gvfs == -1`), then, and *only* then will `gvfs_config_is_set(-1)` return `true`. In all other cases (i.e. if there is even a single unset bit), it will return `false`.